### PR TITLE
Update lifecycle status

### PR DESCRIPTION
**Pull request by nicholasSUSE**
#### Issue:

https://github.com/rancher/ecm-distro-tools/issues/445

#### Solution:

Preparing `charts-build-scripts` command `lifecycle-status` to be migrated into automation inside: [ecm-distro-tools](https://github.com/rancher/ecm-distro-tools)

Changing main behavior of `lifecycle-status` from creating and cloning a temporary directory structure for `rancher/charts repository` to executing all actions in the current charts repository and persisting the results to parsable `state.json` file that can be loaded by different commands. 

#### Result example:

```
{
  "state_file": "/home/nick/WORK/SOFTWARE/go/src/github.com/release-work/charts/state.json",
  "in_lifecycle_current_branch": {
    "elemental": [
      {
        "Version": "104.1.1+up1.5.4"
      },
      {
        "Version": "104.1.0+up1.5.3"
      }
    ],
    "elemental-crd": [
      {
        "Version": "104.1.1+up1.5.4"
      },
      {
        "Version": "104.1.0+up1.5.3"
      }
    ]
  },
  "out_lifecycle_current_branch": {
    "prometheus-federator": [
      {
        "Version": "3.0.0+up0.3.3"
      },
      {
        "Version": "2.1.0+up0.3.0"
      }
    ]
  },
  "released_in_lifecycle": {
    "elemental": [
      {
        "Version": "104.1.1+up1.5.4"
      },
      {
        "Version": "104.1.0+up1.5.3"
      }
    ],
    "elemental-crd": [
      {
        "Version": "104.1.1+up1.5.4"
      },
      {
        "Version": "104.1.0+up1.5.3"
      }
    ]
  },
  "not_released_out_lifecycle": {
    "fleet": [
      {
        "Version": "100.2.3+up0.5.3"
      },
      {
        "Version": "100.2.0+up0.5.1"
      },
      {
        "Version": "100.1.1+up0.4.1"
      },
      {
        "Version": "100.1.0+up0.4.0"
      }
    ]
  },
  "not_released_in_lifecycle": {},
  "released_out_lifecycle": {
    "rancher-backup": [
      {
        "Version": "3.0.0"
      },
      {
        "Version": "2.1.5"
      },
      {
        "Version": "2.1.4"
      }
    ]
  },
  "to_be_released": {},
  "to_be_forward_ported": {}
}
```

### CODE REVIEW

File: pkg/lifecycle/state.go [hunk 1]
```diff
@@ -0,0 +1,136 @@
+package lifecycle
```
Comment (briandowns) [inline pkg/lifecycle/state.go:1]: Remove all logging from library code please.

File: pkg/lifecycle/state.go [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+		return nil, err
+	}
+	if !exist {
+		err = fmt.Errorf("state file does not exist")
+		logrus.Error(err)
+		return nil, err
+	}
+
+	file, _ := os.Open(s.StateFile)
+	// Read the file content
```
Comment (briandowns) [inline pkg/lifecycle/state.go:63]: Move this comment above the operation

File: pkg/lifecycle/state.go [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+	}
+
+	s.StateFile = stateFilePath
+	return nil
+}
+
+// initState will create a new state file and save the state to it
+func (s *Status) initState() error {
+
+	err := s.createStateFile()
```
Comment (briandowns) [inline pkg/lifecycle/state.go:121]: Make all calls that do a single error check inline.

File: pkg/lifecycle/state.go [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+)
+
+// SaveState will save the lifecycle-status state to state.json file at charts repo
+func (s *Status) SaveState() error {
+	logrus.Info("saving app state to state.json file")
+
+	// Marshal the Project struct into JSON
+	data, err := json.MarshalIndent(s, "", "  ")
+	if err != nil {
+		err = fmt.Errorf("error marshalling project to JSON: %w", err)
```
Comment (briandowns) [inline pkg/lifecycle/state.go:23]: Are all of the errors here lacking enough context to warrant having more info added? Usually not necessary and not a standard for the lang or the repo.

**Review by briandowns — CHANGES_REQUESTED** [review]

**Review by tashima42 — APPROVED** [review]

**Comment by nicholasSUSE** [comment]
@briandowns all fixes implemented.

**Review by briandowns — APPROVED** [review]

**Pull request merged**